google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/AddCloudLibrariesWizardAction.java

```diff
@@ -115,24 +120,24 @@ protected String getHelpID() {
      * Overrides {@link DialogWrapper#doOKAction()} to first check if there are any APIs to enable
      * on GCP.
      *
-     * <p>If so, the {@link CloudApiManagementDialog} is opened allowing the user to select a cloud
-     * project and confirm. If the user cancels, the user is returned to this parent dialog.
+     * <p>If so, the {@link CloudApiManagementConfirmationDialog} is opened confirming the API
+     * changes to be made. If the user cancels, the user is returned to this parent dialog.
      * Otherwise, it is closed and the default {@link DialogWrapper#doOKAction()} is invoked.
      */
     @Override
     protected void doOKAction() {
+      CloudProject cloudProject = getCloudProject();
       Set<CloudLibrary> apisToEnable = getApisToEnable();
 
-      if (!apisToEnable.isEmpty()) {
-        CloudApiManagementDialog managementDialog = new CloudApiManagementDialog(project);
+      if (cloudProject != null && !apisToEnable.isEmpty()) {
+        CloudApiManagementConfirmationDialog managementDialog =
+            new CloudApiManagementConfirmationDialog(project, apisToEnable);
         DialogManager.show(managementDialog);
 
         if (managementDialog.isOK()) {
           ProgressManager.getInstance()
               .runProcessWithProgressSynchronously(
-                  () ->
-                      CloudApiManager.enableApis(
-                          apisToEnable, managementDialog.getCloudProject(), project),
+                  () -> CloudApiManager.enableApis(apisToEnable, cloudProject, project),
```

Comment:
We call `runProcessWithProgressSynchronously` with `true /*canBeCanceled*/`, but I cannot see anything that is done on cancel, can it truly cancelled as of now?

Reply:
let me check how this works. I thought that if canceled, than the process is terminated. But since its not set up as a transaction, this really wouldn't make sense

Reply:
yea - good catch - canceling doesn't do anything now. I'll look into supporting it.